### ISSUE TITLE
FBSDK does not allow me to sign in different account after logout

### ISSUE THREAD

**Issue by danish519** [facebook/react-native-fbsdk#712]
when i login to fbsdk it's work fine then i want to logOut which i used to `LoginManager.logOut()` that's what they mentioned to remove user cache . when i click facebook button again i'ts picks my previous account of facebook weather i login in web or app "and say continue with Danish Hussain". it should not ask me to login with another account.

**Comment by BanidAzin** [comment]
I'm also having the same issue. Anyone found any solution for this ?

**Comment by SMKH-PRO** [comment]
same issue

**Comment by ratnani1996** [comment]
Same issue

**Comment by jayawasthi** [comment]
same here

**Comment by crzycoder** [comment]
@ratnani1996 use this
`logOut = (access_token) => {
        let logOut = new GraphRequest(
            "me/permissions/",
            {
                accessToken: access_token,
                httpMethod: 'DELETE'
            },
            (error, result) => {
                console.log('LogOut Result---->>>', result);
                if (error) {
                    console.log('Error fetching data: ' + error.toString());
                } else {
                    LoginManager.logOut();
                }
            });
        new GraphRequestManager().addRequest(logOut).start();
    };`

**Comment by kishansbs** [comment]
@crzycoder  also using this not able to change account.

**Comment by crzycoder** [comment]
@kishansbs changing account is not possible in latest `react-native-fbsdk` .
the method i mentioned is for select account again from facebook. it's select account only which you are sign in in `facebook` app which is currently login.if you change account on `facebook` app the sdk shows that account.

**Comment by ratnani1996** [comment]
@crzycoder thanks for the response, but it didn't work for me. I am unable to change my facebook account.

**Comment by robyweb** [comment]
Has anyone found a solution for this? We can't get our Facebook login approved because the Facebook testers keep trying to sign in with an account from a previous session.

**Comment by mattdamon108** [comment]
Does anybody have a solution? I can't remember the moment exactly, but I could switch the account after I changed the account in the safari in iOS simulator. But I doesn't work now.

**Comment by livnunes** [comment]
Anyone found a solution for this, I'm currently using:
- react-native@0.59.10
- react-native-fbsdk@0.10.1
and ios sdk is at 5.6.0.
Everytime I logout and try to login from a different account I'm unable to do so.

**Comment by SmartArray** [comment]
We have the same "issue". I did some investigations and I can tell you why it's impossible to get a logout working properly. It's not facebook's fault. Rather it is a security measure from Apple.

## Here is why:

FBSDKCoreKit is using the most recent and recommended API provided by Apple: `ASWebAuthenticationSession`.
It provides a secure and scoped session to login to third-party services, offering a secure callback to pass data back to the app.

As the `ASWebAuthenticationSession` is capsuled from the app, the biggest implications are: The app is unable to read/modify/delete the session data (passwords and login data) from the hardened WebView.

Here you can see the API call:

https://github.com/facebook/facebook-ios-sdk/blob/d5dc7380ddc0ea07fa4472c8384491cb366a3ae4/FBSDKCoreKit/FBSDKCoreKit/Internal/BridgeAPI/FBSDKBridgeAPI.m#L367

I thought of possible workarounds. For instance, using the same authentication session store to call the logout endpoint. However, the API would request this alert, which is an inappropriate use case:
<img width="237" alt="Apples ASWebAuthenticationSession Sign In Alert " src="https://user-images.githubusercontent.com/934592/83939574-60fd1b00-a7de-11ea-8b97-a50519a0d9f5.png">

_IMHO, there are only two methods to solve this issue:_

#### Possible Workaround by Facebook 

Either Facebook solves this by integrating a logout button in their OAuth modal form:
<img width="353" alt="Facebook Oauth Form 'Sign In with another account' button proposal" src="https://user-images.githubusercontent.com/934592/83939686-28117600-a7df-11ea-8127-bfcd903a80e9.png">

#### Solution by Apple

Alternatively, Apple provides a secure way to clear cookies or web data from `ASWebAuthenticationSession`. Obviously, this wouldn't leak any privacy information at all, it would just interfere with the Single-Sign-On Feature. However, they could scope the Session Deletion with the URL that is being called by the API.
Just saying.

--- 

**Tl;dr**
Currently, there is no way to do achieve what we want.

**Comment by omer-to** [comment]
I’m having the same issue. Log in dialog never prompts for email and password once the user logged in.
Revoking permissions does no good.

Is there a workaround? I feel like it’s a big issue giving the user impression that he/she will and can never logs out.

Moreover, this logging out behavior might result in rejection in app review, and I don’t think it is fair.

**Comment by LFMAKER** [comment]
Anyone found any solution to this problem?

**Comment by kyle-ssg** [comment]
Assuming noone solved this?  I'm guessing it's related to this https://github.com/facebookarchive/facebook-swift-sdk/issues/215

**Comment by LFMAKER** [comment]
Anyone found any solution to this problem?

**Comment by SahilPawar1994** [comment]
if you are working on an iOS simulator and facing this issue, then you can reset your simulator using command **xcrun simctl erase <device UDID>** or **xcrun simctl erase all** this will erase all the data from your simulator, then run the app again, then you can do the login from another account.

**Comment by victorbruce** [comment]
> xcrun simctl erase all

So far, this is the only solution that has helped me change the account to signing to Facebook. Thanks, @SahilPawar1994

**Comment by saitejamalapati** [comment]
Do we have any work around or any solution for this issue. We have tried clearing the cookies using @react-native-community/cookies when the user logs out of the app. But this didn't work in our case.

**Comment by DangKhoi1997** [comment]
Any updates guys ???

**Comment by pramitigupte** [comment]
Did anyone find the solution?
